apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -281,6 +282,36 @@ const trackBlockReplacement = ( originalBlockIds, blocks, ...args ) => {
  * @returns {void}
  */
 const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
+	// Template Parts and Reusable Blocks are asynchronously loaded blocks.
+	// Content is fetched from the REST API so the inner blocks are
+	// populated when the response is received. We want to ignore
+	// `replaceInnerBlocks` action calls when the `innerBlocks` are replaced
+	// because the template part or reusable block just loaded.
+	const parentBlock = select( 'core/block-editor' ).getBlocksByClientId( rootClientId )?.[ 0 ];
+	if ( parentBlock ) {
+		const { name, innerBlocks } = parentBlock;
+		const isAsyncLoadedEntityBlock =
+			// Template Part
+			name === 'core/template-part' ||
+			// Reusable BLock
+			name === 'core/block';
+		const hasInnerBlocks = innerBlocks.length > 0;
+
+		if ( isAsyncLoadedEntityBlock && ! hasInnerBlocks ) {
+			return;
+		}
+	}
+
+	// Performing an undo or redo will cause affected template parts
+	// and reusable blocks to synchronize their inner blocks.
+	// Which results in `replaceInnerBlocks` calls. We ignore them
+	// to avoid tracking these changes since they are behind the scenes changes.
+	const msElapsedSinceLastUndoOrRedo = Date.now() - lastUndoOrRedoTimestamp;
+	const IGNORE_THRESHOLD_IN_MS = 250;
```

Comment:
Do we know for sure that 250ms is enough for the synchronization of inner blocks to complete? Is there a better way to guarantee that happened?

Reply:
I think a better way to guarantee it is just to ignore all the `replaceInnerBlocks`. I did some digging and it seems to be a good way for now. I'm not sure about its future-proof-ness though. We are very limited here and we can't differentiate between a user action vs behind the scenes action. So we better ignore everything and put some actions on the allow list later on if we need to.

Reply:
Updated the PR description to reflect the latest commits.

Reply:
>I think a better way to guarantee it is just to ignore all the replaceInnerBlocks.

Just noting (as is in the PR description as well) that this is only ignored for Template Parts and Reusable blocks.  Very well described comments in the code 😁 .